component-api/src/main/java/org/talend/sdk/component/api/record/Schema.java

```diff
@@ -60,16 +66,41 @@ default Entry getEntry(final String name) {
     String getProp(String property);
 
     enum Type {
-        RECORD,
-        ARRAY,
-        STRING,
-        BYTES,
-        INT,
-        LONG,
-        FLOAT,
-        DOUBLE,
-        BOOLEAN,
-        DATETIME
+        RECORD(new Class<?>[] { Record.class }),
+        ARRAY(new Class<?>[] { Collection.class }),
+        STRING(new Class<?>[] { String.class }),
+        BYTES(new Class<?>[] { byte[].class, Byte[].class }),
+        INT(new Class<?>[] { Integer.class }),
+        LONG(new Class<?>[] { Long.class }),
+        FLOAT(new Class<?>[] { Float.class }),
+        DOUBLE(new Class<?>[] { Double.class }),
+        BOOLEAN(new Class<?>[] { Boolean.class }),
+        DATETIME(new Class<?>[] { Long.class, Date.class, Temporal.class });
+
+        /** All compatibles Java classes */
+        private final Class<?>[] classes;
+
+        Type(final Class<?>[] classes) {
+            this.classes = classes;
+        }
+
+        /**
+         * Check if input can be affected to an entry of this type.
+         * 
+         * @param input : object.
+         * @return true if input is null or ok.
+         */
+        public boolean isCompatible(final Object input) {
+            if (input == null) {
+                return true;
+            }
```

Comment:
just a remark, we have the `nullable` in entry. checking rest of code...

Reply:
Here, it's just to check compatibility with Schema.Type enum (as instanceof), so null is OK